contributingGuides/TYPESCRIPT_STYLE.md

```diff
@@ -0,0 +1,857 @@
+    .filter(user => user.age >= minAge && user.active === isActive)
+    .map(user => `${user.name} (${user.age})`);
+};
+```
+
+## Const assertions
+
+Const assertions allow you to get more specific types for your constants. TypeScript will infer the most specific type for literals, making them `readonly` and removing wider types. 
+
+Always use const assertions when declaring global constant values.
+
+```ts
+// Bad
+const CONST = {
+    // ...
+};
+
+// Good
+const CONST = {
+    // ...
+} as const;
+```
+
+## Types during error handling
+
+Errors in try/catch clauses are typed as `unknown`, if the developer needs to use the error data they must conditionally check the type of the data first. Use `instanceof` to ensure your error object is an `Error` before accessing its properties.
+
+```ts
+try {
+    // your code
+} catch (e) { // At this point, the type of `e` is `unknown`.
+    if (e instanceof Error) {
+        // Now I can safely access all properties from `e` because
+        // I ensured `e` is an `Error` object.
+        console.error(e.message);
+    }
+}
+```
+
+### Custom type guards
+
+A type guard is a check that narrows the type of a variable within a certain scope.
+
+The `typeof` keyword must be used when you need to narrow the type to primitive structures, like `string` or `number`.
+
+```ts
+function padLeft(value: string, padding: string | number) {
+    if (typeof padding === "number") {
+        return Array(padding + 1).join(" ") + value;
+    }
+    if (typeof padding === "string") {
+        return padding + value;
+    }
+    throw new Error(`Expected string or number, got '${padding}'.`);
+}
+
+padLeft("Hello world", 4); // returns "    Hello world"
+```
+
+We can also create user-defined type guards which are functions that perform a runtime check that guarantees the type in a certain scope. 
+
+```ts
+type PolicyRoomReport = {
+    type: 'policyRoom';
+    id: string;
+    roomId: string;
+};
+
+type PolicyExpenseChatReport = {
+    type: 'policyExpenseChat';
+    id: string;
+    expenseChatId: string;
+};
+
+type ChatReport = PolicyRoomReport | PolicyExpenseChatReport;
+
+function isPolicyRoomReport(report: ChatReport): report is PolicyRoomReport {
+    return report.type === 'policyRoom';
+}
+
+if (isPolicyRoomReport(report)) {
+    // I'm guaranteed to have access to PolicyRoomReport's properties e.g. "roomId".
+    // report.roomId
+} else {
+    // If this is not a PolicyRoomReport, then I'm guaranteed that this is a PolicyExpenseChatReport, so I'll have access to it's properties e.g. "expenseChatId"
+    // report.expenseChatId
+}
+```
+
+In this example, we have the `ChatReport` type which can be two possible types of object, `PolicyRoomReport` or `PolicyExpenseChatReport`. The `isPolicyRoomReport` function will accept `report` as a parameter and will narrow the type of `report` to `PolicyRoomReport` if the condition inside it is true, and to PolicyExpenseChatReport if the condition is false.
```

Comment:
Super nitpick: add backticks to the last usage of PolicyExpenseChatReport to display it as code block